src/background/services/openPayments.ts

```diff
@@ -298,13 +326,60 @@ export class OpenPaymentsService {
       assetScale: walletAddress.assetScale
     })
 
+    await this.initClient(walletAddress.id)
+    await this.completeGrant(amount, walletAddress, recurring)
+
+    await this.storage.set({
+      walletAddress,
+      rateOfPay,
+      minRateOfPay,
+      maxRateOfPay,
+      connected: true
+    })
+  }
+
+  async addFunds({ amount, recurring }: AddFundsPayload) {
+    const { walletAddress } = await this.storage.get(['walletAddress'])
+
+    const grantDetails = await this.completeGrant(
+      amount,
+      walletAddress!,
+      recurring
+    )
+
+    // cancel existing grants of same type, if any
+    const grants = await this.storage.get(['oneTimeGrant', 'recurringGrant'])
+    if (grants.oneTimeGrant && !recurring) {
+      await this.cancelGrant(grants.oneTimeGrant.continue)
+    } else if (grants.recurringGrant && recurring) {
+      await this.cancelGrant(grants.recurringGrant.continue)
+    }
```

Comment:
Isn't this going to cancel the new grant? If `completeGrant` already updated the storage with the new one, we are going to retrieve the grant from storage and cancel them immediately.

I feel like the logic should be:
1. retrieve current grant from storage based on `recurring`;
2. call `completeGrant` - which already performs `.set()` on the storage;
3.  cancel previous grant (if it exists) - we already have the details since we retrieved them at step 1

Reply:
Oops! great catch!